core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -452,6 +452,28 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         return _getCollateral(rawTotalPositionCollateral);
     }
 
+    function _reduceSponsorPosition(address sponsor, FixedPoint.Unsigned memory ratio) internal {
+        // Rounding issues.
+        if (ratio.rawValue == 1) {
+            _deleteSponsorPosition(sponsor);
+            return;
+        }
+
+        PositionData storage positionData = _getPositionData(sponsor);
+        FixedPoint.Unsigned memory collateralToRemove = _getCollateral(positionData.rawCollateral).mul(ratio);
+        FixedPoint.Unsigned memory withdrawalAmountToRemove = positionData.withdrawalRequestAmount.mul(ratio);
+        FixedPoint.Unsigned memory tokensToRemove = positionData.tokensOutstanding.mul(ratio);
```

Comment:
These look like repeated calculations from `createLiquidation`, for the sake of reducing compounded precision-loss errors, could you pass in `liquidatedCollateral` here instead?